filesystem/iso9660.ksy

```diff
@@ -5,269 +5,607 @@ meta:
+      - id: expiration
+        type: b1
+      - id: backup
+        type: b1
+      - id: attributes
+        type: b1
+      - id: access
+        type: b1
+      - id: modify
+        type: b1
+      - id: creation
+        type: b1
+      - id: datetime_short
+        type: rrip_tf_short
+        if: not long_form
+      - id: datetime_long
+        type: rrip_tf_long
+        if: long_form
+  susp_unknown: # default for now
+    seq:
+      - id: length
         type: u1
-      - id: day
+      - id: version
         type: u1
-      - id: hour
+      - id: data
+        size: length - 4
+  su_header:
+    seq:
+      - id: signature
+        type: u2be
+        enum: su_signature
+      - id: su_specific
+        type:
+          switch-on: signature
+          cases:
+            'su_signature::rras_amiga_specific': susp_unknown # AS
+            'su_signature::susp_continuation_area': susp_unknown # CE
+            'su_signature::rrip_child_link': susp_unknown # CL
+            'su_signature::susp_extensions_reference': susp_unknown # ER
+            'su_signature::susp_extension_selector': susp_unknown # ES
+            'su_signature::rrip_alternate_name': rrip_nm # NM
+            'su_signature::susp_padding_field': susp_unknown # PD
+            'su_signature::rrip_parent_link': susp_unknown # PL
+            'su_signature::rrip_posix_device_number': susp_unknown # PN
+            'su_signature::rrip_posix_file_attributes': rrip_px # PX
+            'su_signature::rrip_relocated_directory': susp_unknown # RE
+            'su_signature::rrip_extensions_in_use_indicator': susp_unknown # RR
+            'su_signature::rrip_sparse_file': susp_unknown # SF
+            'su_signature::rrip_symbolic_link': susp_unknown # SL
+            'su_signature::susp_indicator': susp_unknown # SP
+            'su_signature::susp_terminator': susp_unknown # ST
+            'su_signature::rrip_time_file': rrip_tf # TF
+            'su_signature::rrzf_zisofs': susp_unknown # ZF
+  su_headers:
+    seq:
+      - id: header
+        type: su_header
+        repeat: eos
+  directory_records:
+    doc: |
+      First item "." it points to it self
+      Second item ".." it points to the parent, or also to self if it the root
+    seq:
+      - id: directory_record
+        type: directory_record
+        repeat: until
+        repeat-until: _.len_dr == 0
+  directory_record:
+    doc-ref: ecma-119 9.1
+    seq:
+      - id: len_dr
+        doc-ref: ecma-119 9.1.1
         type: u1
-      - id: minute
+      - id: ext_attr_rec_len
+        doc-ref: ecma-119 9.1.2
         type: u1
-      - id: sec
+        if: len_dr > 0x0
+      - id: location_of_extent
+        doc-ref: ecma-119 9.1.3
+        type: u4bi
+        if: len_dr > 0x0
+      - id: data_len
+        doc-ref: ecma-119 9.1.4
+        type: u4bi
+        if: len_dr > 0x0
+      - id: rec_date_time
+        doc-ref: ecma-119 9.1.5
+        type: datetime_short
+        if: len_dr > 0x0
```

Comment:
Shouldn't it be moved into a subtype?

Reply:
I am thinking about this... it was split in the original, but I did like the output from this one more.
This works great, I just don't know the CPU performance of having multiple if statements.

Reply:
It is done, we now have directory_record_head and directory_record_body